app/src/main/kotlin/at/bitfire/davdroid/servicedetection/HomeSetRefresher.kt

```diff
@@ -159,4 +161,31 @@ class HomeSetRefresher @AssistedInject constructor(
         }
     }
 
+    /**
+     * Evaluates whether a response if personal or not.
+     * It takes the [Owner] property from the response, and compares its value against [service]'s [Service.principal].
+     *
+     * If either one of those is not set (`null`), this function returns `null`.
+     * @param davResponse The response to process.
+     * @return
+     * - `true` if the owner matches a principal.
+     * - `false` if the owner doesn't match a principal.
+     * - `null` if the owner and/or principal is not set / unknown.
+     */
+    private fun isPersonal(davResponse: Response): Boolean? {
```

Comment:
I don't think we need to account for the difference for the principal being unknown/not set versus existing, but not matching. The only thing we care about is whether the home set is personal, or not. So how about always returning false in both cases?